File: LV0/연속된 수의 합.js
function solution(num, total) {
  let result = [];

  let average = Math.floor(total / num);

  let middle = Math.floor(num / 2);

  for (let i = average - middle; i <= average + middle; i++) {
    let sum = 0;
    for (let j = i; j < i + num; j++) {
      sum += j;
    }

    if (sum === total) {
      for (let k = i; k < i + num; k++) {
        result.push(k);
      }
      break;
    }
  }

  return result;
}

File: LV0/연속된 수의 합.java
class Solution {

  public int[] solution(int num, int total) {
    int[] result = new int[num];
    int average = total / num;
    int middle = num / 2;
    for (int i = average - middle; i <= average + middle; i++) {
      int sum = 0;
      for (int j = i; j < i + num; j++) {
        sum += j;
      }
      if (sum == total) {
        for (int k = i; k < i + num; k++) {
          result[k - i] = k;
        }
        break;
      }
    }
    return result;
  }
}
